amp.php

```diff
@@ -128,7 +137,7 @@ function amp_maybe_add_actions() {
 	global $wp_query;
 	$post = $wp_query->post;
 
-	$supports = post_supports_amp( $post );
+	$supports = post_supports_amp( $post ) && ! amp_is_canonical();
```

Comment:
Might as well move the `! amp_is_canonical()` check up to the top of the function:

```php
if ( amp_is_canonical() || ! is_singular() || is_feed() ) {
```